frontend/src/components/Public.vue

```diff
@@ -0,0 +1,345 @@
+
+    },
+    routePolyLines(): L.Polyline[] {
+      return this.$store.getters.getRoutePolyLines;
+    },
+    renderRoutes() {
+      if (this.routePolyLines().length > 0 && !this.initialized) {
+        if (this.Map !== undefined && !this.$store.getters.getBoundsPolyLine.isEmpty()) {
+          this.initialized = true;
+          this.Map.fitBounds(this.$store.getters.getBoundsPolyLine.getBounds());
+        }
+      }
+      this.existingRouteLayers.forEach((line) => {
+        if (this.Map !== undefined) {
+          this.Map.removeLayer(line);
+        }
+      });
+      this.existingRouteLayers = new Array<L.Polyline>();
+      this.routePolyLines().forEach((line: L.Polyline) => {
+        if (this.Map !== undefined) {
+          this.Map.addLayer(line);
+          this.existingRouteLayers.push(line);
+        }
+      });
+    },
+    renderStops() {
+      this.$store.state.Stops.forEach((stop: Stop) => {
+        const marker = L.marker([stop.latitude, stop.longitude], {icon: StopIcon});
+        if (this.Map !== undefined) {
+          marker.bindPopup(stop.name);
+          marker.addTo(this.Map);
+        }
+      });
+    },
+    addVehicles() {
+      this.$store.state.Vehicles.forEach((veh: Vehicle) => {
+        if (this.Map !== undefined) {
+          veh.addToMap(this.Map);
+        }
+      });
+    },
+    showUserLocation() {
+      const userIcon = L.icon({
+        iconUrl: 'static/images/user.svg',
+
+        iconSize:     [12, 12], // size of the icon
+        iconAnchor:   [6, 6], // point of the icon which will correspond to marker's location
+        shadowAnchor: [6, 6],  // the same for the shadow
+        popupAnchor:  [0, 0], // point from which the popup should open relative to the iconAnchor
+      });
+
+      if (navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition(((position) => {
+          const locationMarker = {
+                name: 'You are here',
+                marker: L.marker([position.coords.latitude, position.coords.longitude], {
+                    icon: userIcon,
+                    zIndexOffset: 1000,
+                }),
+          };
+          locationMarker.marker.bindPopup(locationMarker.name);
+          if (this.Map !== undefined) {
+            locationMarker.marker.addTo(this.Map);
+          }
+        }) as PositionCallback);
+      }
+    },
+  },
+  components: {
+    dropdown,
+    messagebox,
+  },
+});
+</script>
+
+<style lang="scss">
+.caution-circle {
+  float:right;
+  width:10px;
+  height:10px;
+  background-color:orange;
+  border-radius:50%;
+}
+.pulsate {
+  float:right;
+  width:10px;
+  height:10px;
+  background-color:blue;
+  border-radius:50%;
+  animation: pulsate 2.5s ease-out;
+  animation-iteration-count: infinite;
+}
+@keyframes pulsate {
+  0% {
+    opacity: 0;
+  }
+  50% {
+    opacity: 1;
+  }
+  100% {
+    opacity: 0;
+
+  }
+}
+.livebox{
+    p {
+      margin-right: 5px;
+    }
+    position: absolute;
+    height: 20px;
```

Comment:
It looks a little squished
```suggestion
    height: 26px;
```